src/gtc/dace/transformations.py

```diff
@@ -0,0 +1,112 @@
+import dace
+from dace.transformation.dataflow import TrivialMapElimination
+from dace.transformation.helpers import nest_state_subgraph
+from dace.transformation.interstate import InlineTransients
+
+
+def eliminate_trivial_maps(sdfg: dace.SDFG):
+    """Remove maps and map ranges where the iteration is over a single index."""
+    applied = True
+    while applied:
+        applied = False
+        for map_entry, state in sdfg.all_nodes_recursive():
+            if isinstance(map_entry, dace.nodes.MapEntry):
```

Comment:
If this were made a separate Iterable, then the indentation from the check on line 13 could be removed (this has a lot of levels of indentation already :).
```suggestion
        map_entry_and_state = (x for x in sdfg.all_nodes_recursive() if isinstance(x[0], dace.nodes.MapEntry))
        for map_entry, state in map_entry_and_state:
```

Reply:
i moved all three levels of conditionals into a separate filter method. That way it's even 2 levels shallower and the `continue` statement is also avoided.

Reply:
Never mind, I redid the entire thing using dace transformation API, not sure why I thought this wasn't possible before.